### PR TITLE
LMSDEV-148

### PR DESCRIPTION
**Pull request by davidherzlos**

### CODE REVIEW

File: block_course_modulenavigation.php [hunk 1]
```diff
@@ -324,9 +324,20 @@ public function get_content() {
...
+                    if (!$module->visible || !$module->visibleoncoursepage) {
                         continue;
                     }
+
+                    if (!$module->uservisible) {
+                        if (get_config(
+                                'block_course_modulenavigation',
+                                'toggleshowrestricted') == 1 ) {
+                            continue;
+                        }
```
Comment (petermAprende) [inline block_course_modulenavigation.php]: I found this confusing:
In PHP: 1 usually means true, so this reads to me as:
"If toggleshowrestricted is true, skip". It wasn't until I got to the settings page that I noticed that you had set No == 1. I would recommend changing to use a checkbox in the settings page to get a boolean value.

File: scss/course_modulenavigation.scss [hunk 2]
```diff
@@ -1,280 +1,118 @@
...
+    .section-group {
+        background: $deep-cove;
+        padding: 10px;
     }
-
-    .pagination > li,
-    .allpagination > li {
-        display: table-cell;
+    // Collapsed sections
+    .section {
```
Comment (petermAprende) [inline scss/course_modulenavigation.scss]: Is this selector specific enough? I very much could be wrong, but I feel like we have run into issues in the past where our selectors were not specific enough and ended up selecting sections that we didn't intend.

**Review by petermAprende — CHANGES_REQUESTED** [review]
The change I care about is changing the admin setting to use admin_setting_configcheckbox. The CSS selector caught my attention, but I don't feel like I am knowledgeable enough to request any specific changes.

File: scss/course_modulenavigation.scss (hunk 2 above)
Comment (davidherzlos) [reply scss/course_modulenavigation.scss]: Yes, it is specific, it is actually a descendant selector using the scss syntax.

File: block_course_modulenavigation.php (hunk 1 above)
Comment (davidherzlos) [reply block_course_modulenavigation.php]: Ok, makes sense!

File: settings.php [hunk 3]
```diff
@@ -137,4 +137,31 @@
...
+    ];
+    $settings->add(
+        new admin_setting_configselect(
+            $name,
+            $title,
+            $description,
+            $default,
+            $choices
+        )
+    );
```
[the remark this answers is not in the document]
Comment (davidherzlos) [reply settings.php]: Ok, makes sense!

**Pull request merged**